src/rpc/handlers/GetAggregatePrice.hpp

```diff
@@ -155,14 +164,26 @@ class GetAggregatePriceHandler {
                 return MaybeError{};
             }};
 
+        // validate quoteAsset in accordance to the currency code found in XRPL doc:
+        // https://xrpl.org/docs/references/protocol/data-types/currency-formats#currency-codes
+        static auto const validateQuoteAsset =
+            validation::CustomValidator{[&](boost::json::value const& value, std::string_view) -> MaybeError {
+                ripple::Currency currency;
+                if (!value.is_string() || !(value.get_string().size() == isoCodeLength) ||
+                    !ripple::to_currency(currency, boost::json::value_to<std::string>(value)))
+                    return Error{Status{RippledError::rpcINVALID_PARAMS}};
+
+                return MaybeError{};
+            }};
+
         static auto const rpcSpec = RpcSpec{
             {JS(ledger_hash), validation::Uint256HexStringValidator},
             {JS(ledger_index), validation::LedgerIndexValidator},
             // note: Rippled's base_asset and quote_asset can be non-string. It will eventually return
             // "rpcOBJECT_NOT_FOUND". Clio will return "rpcINVALID_PARAMS" if the base_asset or quote_asset is not a
             // string. User can clearly know there is a mistake in the input.
             {JS(base_asset), validation::Required{}, validation::Type<std::string>{}},
-            {JS(quote_asset), validation::Required{}, validation::Type<std::string>{}},
+            {JS(quote_asset), validation::Required{}, validateQuoteAsset},
```

Comment:
Maybe we can make it simpler by calling :
meta::WithCustomError{validation::CurrencyValidator, Status(RippledError::InvalidParam)}